lib/Sema/CSBindings.cpp

```diff
@@ -376,8 +376,28 @@ ConstraintSystem::getPotentialBindings(TypeVariableType *typeVar) {
         break;
 
       auto type = binding->BindingType;
-      if (exactTypes.insert(type->getCanonicalType()).second)
+      if (exactTypes.insert(type->getCanonicalType()).second) {
         result.addPotentialBinding(*binding);
+
+        if (auto *locator = typeVar->getImpl().getLocator()) {
+          auto path = locator->getPath();
+          auto voidType = TupleType::getEmpty(getASTContext());
```

Comment:
`ASTContext` has `TheEmptyTupleType` for this (and it's already a `CanType`).

Reply:
Included the fix to https://github.com/apple/swift/pull/15187